src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionXpathSingleFilter.java

```diff
@@ -0,0 +1,90 @@
+package com.puppycrawl.tools.checkstyle.filters;
+
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+import com.puppycrawl.tools.checkstyle.api.*;
+
+public class SuppressionXpathSingleFilter extends AutomaticBean implements
+        Filter, ExternalResourceHolder {
+
+    /** The regexp to match check names against. */
+    private Pattern checkRegexp;
+
+    /** The pattern for check class names. */
+    private String checkPattern;
+
+    /** The regexp to match message names against. */
+    private Pattern messageRegexp;
+
+    /** The pattern for message names. */
+    private String messagePattern;
+
+    /** Module id filter. */
+    private String moduleId;
+
+    /** Xpath query. */
+    private String xpathQuery;
+
+    /** Set of individual suppresses. */
+    private FilterSet filters = new FilterSet();
+
+    public void setCheckRegexp(Pattern checkRegexp) {
+        this.checkRegexp = checkRegexp;
+    }
+
+    public void setCheckPattern(String checkPattern) {
+        this.checkPattern = checkPattern;
+    }
+
+    public void setMessageRegexp(Pattern messageRegexp) {
+        this.messageRegexp = messageRegexp;
+    }
+
+    public void setMessagePattern(String messagePattern) {
+        this.messagePattern = messagePattern;
+    }
+
+    public void setModuleId(String moduleId) {
+        this.moduleId = moduleId;
+    }
+
+    public void setXpathQuery(String xpathQuery) {
+        this.xpathQuery = xpathQuery;
+    }
+
+
+    @Override
+    public boolean accept(AuditEvent event) {
+        return filters.accept(event);
```

Comment:
The contents of this method should be the same as https://github.com/checkstyle/checkstyle/blob/6893affeac7285afd16f8a389c7d65023d0c327a/src/main/java/com/puppycrawl/tools/checkstyle/filters/XpathFilter.java#L119-L123 .

Reply:
@jack870131 Another idea instead of duplicating so much code is to make this class a wrapper of `XpathFilter`.
So in the constructor of this class, instantiate `XpathFilter`, make all the setters pass their values to the `XpathFilter` instance, and just have this accept method call `XpathFilter`'s accept.